src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -271,8 +277,12 @@ private Task ProcessAuthentication(bool isAsync = false, bool isApm = false, Can
             }
             finally
             {
-                // Operation has completed.
-                _nestedAuth = 0;
+                if (result == null || result.IsCompleted)
+                {
+                    // Operation has completed.
+                    _handshakeBuffer.Dispose();
```

Comment:
If result completes in failure/cancellation, this could end up running concurrently with the _handshakeBuffer.Dispose; _nestedAuth = 0; in the ContinueWith callback.  That seems wrong, no?